src/app/utils/socket-utils.ts

```diff
@@ -458,9 +461,47 @@ export function getWeaponSockets(
     PlugCategoryHashes.V300WeaponDamageTypeKinetic,
   ];
 
+  let moddedSockets: DimSockets = item.sockets;
+  if (includeFakeMasterwork) {
+    const allSockets = item.sockets.allSockets.map((socket) => {
+      if (socket.socketDefinition.socketTypeHash !== weaponMasterworkY2SocketTypeHash) {
+        return socket;
+      }
+      const mwHash = item.masterworkInfo?.stats?.find((s) => s.isPrimary)?.hash || 0;
+      const plugCategory = D2PlugCategoryByStatHash.get(mwHash);
+      let fullMasterworkPlug =
+        socket.plugSet &&
+        plugCategory &&
+        maxBy(
+          socket.plugSet.plugs.filter((p) => p.plugDef.plug.plugCategoryHash === plugCategory),
+          (plugOption) => plugOption.plugDef.investmentStats[0]?.value,
```

Comment:
Out of curiosity - is the masterwork bonus always the same stat bonus that a tier 10 masterwork would provide on a regular item?

Reply:
From my understanding, I think the bonus varies based on what level the weapon is, with additional stats added at level 20, so not quite.